externals/llvm-external-projects/torch-mlir-dialects/lib/Conversion/TcpToTosa/TcpToTosa.cpp

```diff
@@ -0,0 +1,113 @@
+#include "torch-mlir-dialects/Conversion/TcpToTosa/TcpToTosa.h"
+
+#include "torch-mlir-dialects/Dialect/Tcp/IR/TcpDialect.h"
+#include "torch-mlir-dialects/Dialect/Tcp/IR/TcpOps.h"
+
+#include "../PassDetail.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Pass/PassManager.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/Passes.h"
+
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+
+namespace mlir {
+
+#define GEN_PASS_DEF_CONVERTTCPTOTOSA
+#include "torch-mlir-dialects/Conversion/Passes.h.inc"
+
+namespace torch {
+namespace tcp {
+
+namespace {
+
+template<typename TcpOpT, typename TosaOpT>
+class ConvertElementwiseTcpOpToTosaOp : public OpConversionPattern<TcpOpT> {
+public:
+  using OpConversionPattern<TcpOpT>::OpConversionPattern;
+  using OpAdaptor = typename TcpOpT::Adaptor;
+  LogicalResult matchAndRewrite(
+            TcpOpT op,
+            OpAdaptor adaptor,
+            ConversionPatternRewriter& rewriter) const override {
+    llvm::SmallVector<mlir::Type> newTypes;
+    for (auto res : op->getResultTypes()) {
```

Comment:
Unnecessary `{`.

Reply:
I didn't realized this was the convention being followed in MLIR. I saw both usages actually, so, not quite sure. Anyways, will follow this from now. Thanks for pointing that out.